src/video/sync/merge.ts

```diff
@@ -0,0 +1,80 @@
+import { tmpdir } from 'node:os';
+import { spawnSync } from 'node:child_process';
+import { mkdtempSync, writeFileSync } from 'node:fs';
+import { join } from 'node:path';
+
+import { Syncer, VideoFile } from './types';
+import { Test } from '@saucelabs/sauce-json-reporter';
+
+/**
+ * MergeSyncer is used to synchronize the video start time of a test case with
+ * a collection of video files. Videos are aggregated and their cumulative
+ * runtime is used to mark the video start time of the next test case to be
+ * added.
+ */
+export class MergeSyncer implements Syncer {
+  duration: number;
+  videoFiles: VideoFile[];
+
+  constructor() {
+    this.duration = 0;
+    this.videoFiles = [];
+  }
+
+  public sync(test: Test, video: VideoFile): void {
+    if (video.path && video.duration) {
+      test.videoTimestamp = this.duration / 1000;
+
+      this.videoFiles.push({ ...video });
+      this.duration += video.duration;
+    }
+  }
+
+  public mergeVideos() {
+    if (this.videoFiles.length === 0) {
+      return;
+    }
+    const hasFFMpeg = spawnSync('ffmpeg', ['-version']).status === 0;
+    if (!hasFFMpeg) {
+      console.error(
+        `Failed to merge videos: ffmpeg could not be found. \
+Ensure ffmpeg is available in your PATH.`,
+      );
+      return;
+    }
+
+    const tmpDir = mkdtempSync(join(tmpdir(), 'pw-sauce-video-'));
```

Comment:
Are we responsible for deleting the temp dir? 🤔

Reply:
Good call. I'm honestly not sure the best way to do this. I've attached a cleanup function to process.onExit, but that doesn't get called when the process receives any termination signals. I think that's...fine? I don't want to set my own signal handlers since that clobbers the defaults which doesn't seem ideal.

Reply:
Perfect 👌